### PR TITLE
feat: update ingress controller and ingress class to support test of self-signed TLS certificate

### LINKED ISSUE #1763: add E2E testing for verifying that the passed in signed certificates are being leveraged
We should be able to just create self signed certificates then use `curl` with the option to not verify as a test.  Each certificate test assertion should be in an existing E2E test if possible.  The following need to be verified:
- for `solo network deploy`:  Flags.grpcTlsCertificatePath, Flags.grpcTlsKeyPath, Flags.grpcWebTlsCertificatePath, Flags.grpcWebTlsKeyPath
- for `solo explorer deploy`: we need to create the certificate and use the values file overrides: <https://github.com/hiero-ledger/hiero-mirror-node-explorer/blob/8e38088db6b337b6b5ed17f11cd4643c15c601f7/chart/values.yaml#L63>
- for `solo mirror-node deploy`: it mentions certificates in the README <https://github.com/hiero-ledger/hiero-mirror-node/blob/3ab32072139ff97e02c022ba542765e23dfd8326/charts/README.md?plain=1#L72>, for more details we will need to post a question in the Slack#mirror-node channel

### PR DESCRIPTION
**Pull request by JeffreyDallas**
## Description

This pull request changes the following:

* TBD

### Related Issues

* Closes #1763

### Pull request (PR) checklist

* \[ ] This PR added tests (unit, integration, and/or end-to-end)
* \[ ] This PR updated documentation
* \[ ] This PR added no TODOs or commented out code
* \[ ] This PR has no breaking changes
* \[ ] Any technical debt has been documented as a separate issue and linked to this PR
* \[ ] Any `package.json` changes have been explained to and approved by a repository manager
* \[ ] All related issues have been linked to this PR
* \[ ] All changes in this PR are included in the description
* \[ ] When this PR merges the commits will be squashed and the title will be used as the commit message, the 'commit message guidelines' below have been followed

### Testing

* \[ ] This PR added unit tests
* \[ ] This PR added integration/end-to-end tests
* \[ ] These changes required manual testing that is documented below
* \[ ] Anything not tested is documented

The following manual testing was done:

* TBD

The following was not tested:

* TBD

<details>
<summary>
Commit message guidelines
</summary>
We use 'Conventional Commits' to ensure that our commit messages are easy to read, follow a consistent format, and for automated release note generation. Please follow the guidelines below when writing your commit messages:

1. BREAKING CHANGE: a commit that has a footer BREAKING CHANGE:, or appends a ! after the type/scope, introduces a breaking API change (correlating with MAJOR in Semantic Versioning). A BREAKING CHANGE can be part of commits of any type.  NOTE: currently breaking changes will only bump the MAJOR version.
2. The title is prefixed with one of the following:

| Prefix    | Description                                         | Semantic Version Update | Captured in Release Notes |
|-----------|-----------------------------------------------------|-------------------------|---------------------------|
| feat:     | a new feature                                       | MINOR                   | Yes                       |
| fix:      | a bug fix                                           | PATCH                   | Yes                       |
| perf:     | performance                                         | PATCH                   | Yes                       |
| refactor: | code change that isn't feature or fix               | none                    | No                        |
| test:     | adding missing tests                                | none                    | No                        |
| docs:     | changes to documentation                            | none                    | Yes                        |
| build:    | changes to build process                            | none                    | No                        |
| ci:       | changes to CI configuration                         | none                    | No                        |
| style:    | formatting, missing semi-colons, etc                | none                    | No                        |
| chore:    | updating grunt tasks etc; no production code change | none                    | No                        |

</details>

### CODE REVIEW

File: .github/workflows/script/gcs_test.sh [hunk 1]
```diff
@@ -3,6 +3,32 @@ set -eo pipefail
...
+# $4: Protocol (http/https) for logging
+check_service_health() {
+  local url=$1
+  local expected_content=$2
+  local service_name=$3
+  local protocol=$4
+  local curl_args=()
+
+  if [[ $url == https://* ]]; then
+    curl_args+=(-k)
```
Comment (rbarker-dev) [inline .github/workflows/script/gcs_test.sh:19]: why are we using the -k option? this disable certificate validation on curl, should we be doing that?

File: src/core/helpers.ts [hunk 2]
```diff
@@ -567,3 +569,77 @@ export async function getAppleSiliconChipset(logger: SoloLogger) {
...
+  directory: string,
+  name: string = 'localhost',
+): Promise<{certificatePath: string; keyPath: string}> {
+  // Define attributes for the certificate
+  const attributes: {name: string; value: string}[] = [{name: 'commonName', value: name}];
+  const certificatePath: string = PathEx.join(directory, `${name}.crt`);
+  const keyPath: string = PathEx.join(directory, `${name}.key`);
+
+  // Generate the certificate and key
+  selfsigned.generate(attributes, {days: 365}, (error, pems) => {
```
Comment (instamenta) [inline src/core/helpers.ts:632]: wrap this together with the callback in promise, this would make the code behave as async and would remove the:
```
// wait a few seconds for the files to be created
await new Promise(resolve => setTimeout(resolve, 2000));
```

Comment (instamenta) [inline src/core/helpers.ts:632]: wrap this together with the callback in promise, this would make the code behave as async and would remove the:
```
// wait a few seconds for the files to be created
await new Promise(resolve => setTimeout(resolve, 2000));
```

**Review by instamenta — CHANGES_REQUESTED** [review]

File: .github/workflows/script/gcs_test.sh (hunk 1 above)
Comment (JeffreyDallas) [reply .github/workflows/script/gcs_test.sh:19]: This is to a test self signed certificate, need use "-k" to test https://localhost,
without option, curl would complain about the certificate and abort.

File: src/core/helpers.ts (hunk 2 above)
Comment (JeffreyDallas) [reply src/core/helpers.ts]: Updated

File: test/e2e/commands/network.test.ts [hunk 3]
```diff
@@ -41,11 +41,20 @@ describe('NetworkCommand', function networkCommand() {
   argv.setArg(flags.applicationEnv, applicationEnvironmentFilePath);
   argv.setArg(flags.loadBalancerEnabled, true);
 
+  const tmpDir = os.tmpdir();
```
[the remark this answers is not in the document]
Comment (JeffreyDallas) [reply test/e2e/commands/network.test.ts]: Fixed

File: src/core/helpers.ts [hunk 4]
```diff
@@ -567,3 +569,79 @@ export async function getAppleSiliconChipset(logger: SoloLogger) {
...
+      throw new SoloError('failed to create secret for explorer TLS certificates');
+    }
+  } catch (error: Error | any) {
+    const errorMessage: string =
+      'failed to create secret for explorer TLS certificates, please check if the secret already exists';
+    throw new SoloError(errorMessage, error);
+  }
+}
+
+export async function generateTLS(
```
[the remark this answers is not in the document]
Comment (JeffreyDallas) [reply src/core/helpers.ts]: Fixed

File: src/core/helpers.ts [hunk 5]
```diff
@@ -567,3 +569,79 @@ export async function getAppleSiliconChipset(logger: SoloLogger) {
...
+ * Creates a TLS secret in Kubernetes for the Explorer
+ * @param logger Logger instance
+ * @param k8Factory Kubernetes factory instance
+ * @param namespace Namespace to create the secret in
+ * @param domainName Domain name for the TLS certificate
+ * @param cacheDirectory Directory to store temporary files
+ * @param secretName Name of the secret to create
+ * @returns Promise<boolean> - True if secret was created successfully
+ */
+export async function createTLSSecret(
```
[the remark this answers is not in the document]
Comment (JeffreyDallas) [reply src/core/helpers.ts]: Fixed

**Review by jeromy-cannon — APPROVED** [review]

**Review by jeromy-cannon — APPROVED** [review]
